### DISCUSSION TITLE
Question: Auto-Cpufreq

### DISCUSSION THREAD

**Discussion by sprc9034** [Nitrux/nitrux-bug-tracker#96] [General]
I wanted to install the auto cpufreq tool, that's a cpu governer. I can't find a way to install it through distrobox on the host system as it uses systemd( in this case openrc) to start at boot up. Can you pls guide me through how to install it. Project link(https://github.com/AdnanHodzic/auto-cpufreq). Any help would be appreciated as auto-cpufreq combined with tlp gives me great battery life.

**Comment by UriHerrera** [comment]
You have various choices depending on the distribution you use according to the official [auto-cpufreq repository](https://github.com/AdnanHodzic/auto-cpufreq).

You can **install** `auto-cpufreq` using its Snap, a package from the AUR, or manually.

The repository also lists instructions for installation, and the rest depends on how you installed it.

You can create an Ubuntu container, use the Snap or an Arch container, and install it from the AUR or a container of another distribution (they mention openSUSE, Red HAT, Solus, and Void) and run their installer.

In any case, I guess you want to start `auto-cpufreq` automatically. Do you need `auto-cpufreq` to start during boot (as a service)? Otherwise, you can auto-start it during log-in. We have a [tutorial](https://nxos.org/tutorial/how-to-create-service-files-in-nitrux-pt-2-user/) for that. 

If you're using a container to run `auto-cpufreq` and it requires root permissions, you can follow the example on our [Distrobox tutorial](https://nxos.org/tutorial/how-to-use-distrobox-in-nitrux/) and create an alias to run it with `sudo`.

If you do want to start it as a service and you used a container, we can explore that.

A fourth option is to add the [binary](https://github.com/AdnanHodzic/auto-cpufreq/blob/master/bin/auto-cpufreq) to the root of Nitrux and use the already [available service script for OpenRC](https://github.com/AdnanHodzic/auto-cpufreq/blob/master/scripts/auto-cpufreq-openrc) that its developer provides. You have to disable the immutability to do this. However, the distribution only includes some Python packages listed in the [requirements](https://github.com/AdnanHodzic/auto-cpufreq/blob/master/requirements.txt) on the repository.

In that case, you can create an issue asking for a feature request to add the other Python packages. Or you can download the Debian packages, extract them and copy the contents to the root directory. As before, you have to disable the immutability to do this.

And a fifth option is not to use `auto-cpufreq` and instead edit our script in `/etc/local.d/cpu_governor.start` and change the CPU governor from performance to on-demand unless you want the more advanced features that `auto-cpufreq` may provide.
